python/ray/serve/tests/test_api.py

```diff
@@ -101,6 +113,22 @@ def f():
             endpoint_name, backend="backend", route="/different")
 
 
+def test_reject_duplicate_endpoint_and_route(serve_instance):
+    class SimpleBackend(object):
+        def __init__(self, message):
+            self.message = message
+
+        def __call__(self, *args, **kwargs):
+            return {"message": self.message}
+
+    serve.create_backend("backend1", SimpleBackend, "First")
+    serve.create_backend("backend2", SimpleBackend, "Second")
+
+    serve.create_endpoint("test", backend="backend1", route="/test")
+    with pytest.raises(ValueError):
+        serve.create_endpoint("test", backend="backend2", route="/test")
```

Comment:
Should there be another test case for when the backend name is the same and route is different? Or is that tested elsewhere?

Reply:
Yeah, that case is tested in `test_reject_duplicate_endpoint()`.